osu.Game/Overlays/UserProfileOverlay.cs

```diff
@@ -41,6 +42,18 @@ public UserProfileOverlay()
 
         protected override Color4 BackgroundColour => ColourProvider.Background6;
 
+        public override bool OnPressed(KeyBindingPressEvent<GlobalAction> e)
+        {
+            switch (e.Action)
+            {
+                case GlobalAction.ToggleProfile:
+                    Hide();
+                    return true;
+            }
+
+            return base.OnPressed(e);
+        }
```

Comment:
Please move this to `OsuGame` to keep everything in one place. This kind of cross-dependency makes understanding and ensuring correctness very hard.

You can call `userProfile.Hide()` in the case it is already shown.